core/include/scipp/core/element/sort.h

```diff
@@ -7,49 +7,41 @@
 #include "scipp/common/overloaded.h"
 #include "scipp/common/span.h"
 #include "scipp/core/element/arg_list.h"
+#include "scipp/core/element/comparison.h"
 #include "scipp/core/transform_common.h"
 #include "scipp/core/value_and_variance.h"
 #include "scipp/units/unit.h"
 
-#include <stddef.h>
-
 namespace scipp::core::element {
 
-constexpr auto sort_common = overloaded{
-    core::element::arg_list<span<int64_t>, span<int32_t>, span<double>,
-                            span<float>, span<std::string>>,
-    [](units::Unit &) {}};
-
-template <typename F1, typename F2>
-constexpr auto sort_to_specify(F1 for_zip, F2 for_no_var) {
+namespace {
+template <class Compare> constexpr auto make_sort(Compare compare) {
   return overloaded{
-      sort_common, [&for_zip, &for_no_var](auto &range) {
+      core::element::arg_list<span<int64_t>, span<int32_t>, span<double>,
+                              span<float>, span<std::string>>,
+      [](units::Unit &) {},
+      [&compare](auto &range) {
         using T = std::decay_t<decltype(range)>;
         constexpr bool vars = is_ValueAndVariance_v<T>;
         if constexpr (vars) {
-          using B = typename T::value_type;
-          using A = typename B::value_type;
-          std::vector<std::pair<A, A>> zipped;
+          std::vector<ValueAndVariance<typename T::value_type::value_type>>
```

Comment:
What does the double `::value_type::value_type` refer to in this case?

Reply:
The `ValueAndVariance` contains a span for values and a span for variances. The first `value_type` gives the type of that, e.g., `std::span<double>`. The second one gets the `double`.